### DISCUSSION TITLE
How can we use route parameters with Filament panels?

### DISCUSSION THREAD

**Discussion by myregistration** [filamentphp/filament#12757] [Help]
### Package

Panel builder

### Package Version

v3

### How can we help you?

I need to set a route slug parameter to use a particular model instance with the panel panel.  For example, ``/organization/{organization:slug}/``.  How can we modify Filament routes to include parameters?  Thanks

**Comment by leandrocfe** [comment]
I think you should override the route in you web.php file

**Reply by myregistration** [reply]
I tried that but getting some weird side effects, like it's showing the wrong panel (I have multiple), it's redirecting to the default panel.

**Reply by myregistration** [reply]
Do you know how to override the Filament routes?   I don't think it's straightforward.  Thanks!

**Comment by danieledelgiudice** [comment]
Not sure if you've already tried it, but I had a similar requirement and after trying different things (clusters, nested resources, etc.), I actually solved it using [Multi-tenancy](https://filamentphp.com/docs/3.x/panels/tenancy).

When you specify that a Panel has a tenant model (in your case, I would choose the Organization model), you effectively give it a parameter. You can then access models scoped by relationships with the current tenant via global scopes and/or by accessing the current tenant with `Filament::getTenant()`.

The documentation on that page is very thorough and covers several use cases, and you can customize your setup a bit (e.g., I chose not to have registration/profile pages).

Let us know how it goes!

**Reply by myregistration** [reply]
Thanks for the suggestion.  I considered using Tenancy because it seemed to handle similar aspects, but I don't want to use Teams.  Do you need to use Teams in order to utilize the features of Tenancy?  Thanks

**Reply by danieledelgiudice** [reply]
No, the code snippets in the docs use the `Team` class as a placeholder for the tenant model, but you can use any model you want to.
I guess in your case it would be `Organization`. Every time they mention a "team" just pretend they're referring to your tenant model.

Once you create the 2nd panel, you only need to setup the tenancy for it ([link](https://filamentphp.com/docs/3.x/panels/tenancy#setting-up-tenancy)) and you're mostly done.
You don't need to add a tenant registration / profile page / billing if you don't need to.

By default every resource in the 2nd panel will be scoped by their relationship with the tenant class (you can customize it but they look for one based on the model name, so if you have an `Organization` tenant model they would look for the `organization` relationship).

Several other use cases are covered in the docs, but things are mostly self-explanatory.

**Reply by myregistration** [reply]
I'll give it a try.  I was worried about using something built-in that wasn't intended for it in case there are side effects with future updates.  I wish there was a better solution that followed Laravel basics.  I'll let you know if it works.  Thanks

**Reply by myregistration** [reply]
Multi-tenancy didn't work for my needs.  Any other way to add a slug to the Panel's path?  Maybe using Middleware?  Thanks!

**Reply by bilogic** [reply]
I don't know if this works, but this is what I would try:

`$panel` has a `->path(...)`, what happens when you put `->path('/organization/{path}')`?

However, I suspect it will also require a view composer to resolve `{path}` in which multi-tenancy is used to determine which tenant (organization) should be applied.

**Reply by myregistration** [reply]
Thanks, I had tried that initially.  Would have made my life a lot easier if we could just model bind like that.   I can't use multi-tenancy because I am using a one-to-many, not many-to-many relationship.  My users can be a member of multiple organizations, but they have a member record per each organization, so don't need belongsToMany relationship.  I hope that makes sense.  Starting to regret using Filament, but I've invested too much time at this point.  Hopefully there is a solution.  I'm considering using domains, but I don't really like that option either.  My current workaround is to append the organization's uuid to the URL, but that sucks, too.  Let me know if you can think of anything else.  Thanks!

**Reply by bilogic** [reply]
Your OP seeks a solution, but I can't quite grasp the problem (on why multi-tenancy doesn't fit), it might be more fruitful to give context of the business problem you are trying to solve without using terms like `belongsToMany` etc, because they already imply a solution. Speaking rhetorically, if it is pointing to a solution, why doesn't it fit?

Another idea: if what you want is same panel A for different organizations, then it would be copying filament's `web.php` and wrapping a group around it, i.e. in the form of `/{tenant}/panel-A-id`

**Comment by myregistration** [comment]
I was able to get multi-tenancy working by creating a pivot table for organization_user.  I was avoiding it because I didn't really need it since I have a one-to-many table that stores a user's member record per organization, but that's the only way I could get tenancy to work.  It seems like Filament requires a many-to-many relationship for the multi-tenancy.  I'm using the member model's saved lifecycle event to update the pivot table, so it is low maintenance.  Thanks for your help!
